src/python/pants/backend/sql/lint/sqlfluff/skip_field.py

```diff
@@ -10,8 +14,8 @@ class SkipSqlfluffField(BoolField):
     help = "If true, don't run sqlfluff on this target's code."
 
 
-def rules():
-    return [
+def rules() -> Iterable[UnionRule]:
```

Comment:
This is not an obvious change. All of these are UnionRules? Any reason to narrow the type hint like this instead of `Iterable[Rule]` like the other files?

Reply:
`Rule` doesn't work. 

Generally I would use `Rule | UnionRule` which I could update this to

Reply:
see my other comment about potentially having a dedicated type for this